src/Volume.ts

```diff
@@ -200,14 +209,27 @@ export default class Volume {
     this.normRegionOffset = subregionOffset.clone().divide(volumeSize);
   }
 
+  updateRequiredData(required: Partial<LoadSpec>) {
+    this.loadSpecRequired = { ...this.loadSpecRequired, ...required };
+    // if newly required data is not currently contained in this volume...
+    if (
+      this.loadSpecRequired.time !== this.loadSpec.time ||
+      this.loadSpecRequired.scene !== this.loadSpec.scene ||
+      !this.loadSpec.subregion.containsBox(this.loadSpecRequired.subregion)
+    ) {
+      // ...clone `loadSpecRequired` into `loadSpec` and load
+      this.loadSpec = { ...this.loadSpecRequired, subregion: this.loadSpecRequired.subregion.clone() };
+      this.loader?.loadVolumeData(this);
+    }
```

Comment:
something feels weird about the naming of loadSpec vs loadSpecRequired.   I don't really have a good suggestion for renaming. maybe loadSpecToRequest?
Also, I know it's premature but can't help but start thinking about how this code interacts with cache.  An early naive implementation could assume that cache will only have either single slices or full volume and can just check dimensions+channel for exact matches.